packages/repo-tools/src/lib/exec.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { promisify } from 'util';
+import { ExecOptions, exec as execCb } from 'child_process';
+
+const execPromise = promisify(execCb);
+
+export const exec = (
+  command: string,
+  options: string[] = [],
+  execOptions?: ExecOptions,
+) => {
+  return execPromise(
+    [
+      command,
+      ...options.filter(e => e).map(e => (e.startsWith('-') ? e : `"${e}"`)),
```

Comment:
Hm, manual quoting is a bit unfortunate. Can this be rewritten to use the `spawn('a', ['b', 'c'], { windowsVerbatimArguments: true }` form or so?

Reply:
Thoughts on using the CLI's [run](https://github.com/backstage/backstage/blob/acfd99aefda51c282649a9febdc81bc22f179882/packages/cli/src/lib/run.ts#L39) command here? I think it would require extracting into `cli-node` or `cli-common`, but would come with other benefits like stdout forwarding.